File: lexer/defined/DataTypes.ts
import TokenType from "./TokenType.ts";

export default class DataTypes {
    public static list: TokenType[] = [
        TokenType.Num,
        TokenType.String,
        TokenType.Bool,
        TokenType.Nothing,
        TokenType.StringArray,
        TokenType.NumArray,
        TokenType.BoolArray,
        TokenType.Discrete,
    ];
}

File: lexer/src/data_types.rs
use code::token_type::TokenType;
use lazy_static::lazy_static;

lazy_static! {
    pub static ref DATA_TYPES: Vec<TokenType> = vec![
        TokenType::StringArray,
        TokenType::NumArray,
        TokenType::BoolArray,
        TokenType::String,
        TokenType::Num,
        TokenType::Bool,
        TokenType::Discrete,
        TokenType::Nothing
    ];
}
pub fn is_data_type(token_type: TokenType) -> bool {
    DATA_TYPES.contains(&token_type)
}